src/republication.rs

```diff
@@ -128,6 +110,24 @@ impl<'a> RepublicationBuffer<'a> {
         Ok(())
     }
 
+    fn probe_header(&mut self, index: usize) -> Option<(usize, MqttHeader)> {
+        let offset = self
+            .pending_pub
+            .iter()
+            .take(index)
+            .map(|(_, len)| len)
+            .sum();
+        let (id, len) = self.pending_pub.iter().nth(index).unwrap();
```

Comment:
Interesting! I tried to put this into a single iter with `fold()` but it's much less comprehensible.